### PR TITLE
Adds a toy train to the arcades

### PR DESCRIPTION
**Pull request by Jake-Wheal**
## About The Pull Request

This was made as my very first PR for SS13 so is not very ambitious but nonetheless it adds a small toy train to the game that can be won from the arcades like any other toy, the train plays a sound when interacted with to give it some depth but has no practical purpose other then whatever the purpose of the currently obtainable toys are.

The sprite is by hakewheel (discord: Mylavi#0883)

## Why It's Good For The Game

As I say above this is my first PR so it is never going to add an overwhelming amount to the game but I think its nice for people who play on the arcade machines regularly to find a brand new toy and do whatever people do with toys already.

## Changelog
:cl:
add: Added toy train in arcade machines
/:cl:

### CODE REVIEW

**Comment by badideaguy** [comment]
My first thought was “are there toy train tracks for this to lag the server out with”

**Comment by ghost** [comment]
Babby first PR.

**Comment by ghost** [comment]
The sprite for reference.

image

**Comment by ghost** [comment]
@ike709 We need a good first PR label.

**Comment by zeskorion** [comment]
What's this?  A first pr that is somehow acceptable code?

This cant bee the beestation repo.

**Comment by utilbird** [comment]
choo choo

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1089,6 +1089,25 @@
...
+	desc = "A little toy train for young spess men. The wheels are attached to a piece of string!"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "toytrain"
+	w_class = WEIGHT_CLASS_SMALL
+	var/cooldown = 0
+
+/obj/item/toy/toytrain/attack_self(mob/user)
+	if (cooldown < world.time)
+		cooldown = (world.time + 20) // Sets cooldown at 2 seconds
+		user.visible_message("<span class='warning'>[user] pulls the string on the bottom of the train.</span>", "<span class='notice'>You pull the string, the wheels begin to spin</span>", "<span class='italics'>The train whires audibly.</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/toys.dm]:
```suggestion
		user.visible_message("<span class='warning'>[user] pulls the string on the bottom of the train.</span>", "<span class='notice'>You pull the string, the wheels begin to spin.</span>", "<span class='italics'>The train whires audibly.</span>")
```

**Comment by PowerfulBacon** [comment]
merge conflict

**Comment by Jake-Wheal** [comment]
> merge conflict

What does that mean?

**Comment by PowerfulBacon** [comment]
You need to resolve the conflicts otherwise it can't be merged, someone must have edited toys.dm

**Comment by Jake-Wheal** [comment]
> You need to resolve the conflicts otherwise it can't be merged, someone must have edited toys.dm

I'm not sure how you are meant to do it but I think I may have fixed it, otherwise I'm gonna need some help

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1089,6 +1089,25 @@
...
+	name = "toy train"
+	desc = "A little toy train for young spess men. The wheels are attached to a piece of string!"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "toytrain"
+	w_class = WEIGHT_CLASS_SMALL
+	var/cooldown = 0
+
+/obj/item/toy/toytrain/attack_self(mob/user)
+	if (cooldown < world.time)
+		cooldown = (world.time + 20) // Sets cooldown at 2 seconds
```
Comment (ike709) [inline code/game/objects/items/toys.dm]:
```suggestion
		cooldown = (world.time + 50) // Sets cooldown at 5 seconds
```

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1089,6 +1089,25 @@
...
+		user.visible_message("<span class='notice'>[user] pulls the string on the bottom of the train.</span>", "<span class='notice'>You pull the string, the wheels begin to spin.</span>", "<span class='italics'>The train whires audibly.</span>")
+		playsound(src, 'sound/voice/moth/scream_moth.ogg', 5, 0)
+
```
Comment (ike709) [inline code/game/objects/items/toys.dm:1108]: Add an `else` with a `to_chat()` to the user that it's on cooldown.

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1089,6 +1089,25 @@
...
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "toytrain"
+	w_class = WEIGHT_CLASS_SMALL
+	var/cooldown = 0
+
+/obj/item/toy/toytrain/attack_self(mob/user)
+	if (cooldown < world.time)
+		cooldown = (world.time + 20) // Sets cooldown at 2 seconds
+		user.visible_message("<span class='notice'>[user] pulls the string on the bottom of the train.</span>", "<span class='notice'>You pull the string, the wheels begin to spin.</span>", "<span class='italics'>The train whires audibly.</span>")
+		playsound(src, 'sound/voice/moth/scream_moth.ogg', 5, 0)
```
Comment (ike709) [inline code/game/objects/items/toys.dm:1107]: I don't remember what this sounds like but it's called `scream_moth` so it doesn't seem appropriate for a toy train.

Comment (Jake-Wheal) [reply code/game/objects/items/toys.dm:1107]: The noise is supposed to be the the wheels whirring after the string is pulled, `scream_moth` is the best I could find for it. It sounds similar to the bee plush and I think it is the sound the moth plush uses if you can remember those.

**Pull request closed without merging**